test/abstract_client.js

```diff
@@ -973,6 +973,115 @@ module.exports = function (server, config) {
       }
     })
 
+    it('should handle error with async incoming store in QoS 2 `handlePublish` method', function (done) {
+      function AsyncStore () {
+        if (!(this instanceof AsyncStore)) {
+          return new AsyncStore()
+        }
+      }
+      AsyncStore.prototype.put = function (packet, cb) {
+        setTimeout(function () {
+          cb(new Error('Error'))
+        }, 200)
```

Comment:
can you shorten these? A process.nextTick might be enough.

Reply:
Done.